src/main/java/edu/kis/powp/jobs2d/command/gui/RefuelInkWindow.java

```diff
@@ -0,0 +1,60 @@
+package edu.kis.powp.jobs2d.command.gui;
+
+import edu.kis.powp.appbase.gui.WindowComponent;
+import edu.kis.powp.jobs2d.drivers.InkUsageDriverDecorator;
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+
+public class RefuelInkWindow extends JFrame implements WindowComponent
+{
+    private InkUsageDriverDecorator driver;
```

Comment:
Lack of ISP. RefuelInkWindow only uses restoreInk, but this method is tangled with Jobs2dDriver API.

Reply:
You did not understand my suggestion (you have made many unnecessary changes). ISP (Interface Segregation Principle) issue concerned the driver field. Earlier driver was used only for restoreInk, now it is used only for setInkLimit. What I wanted you to do was to change this line to:
private InkOperator operator
I wanted to add InkOperator interface to InkUsageDriverDecorator so it could be used here without any other changes to RefuelInkWindow or InkUsageDriverDecorator interface.